wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -916,6 +930,15 @@ public void onDestroy() {
                 log.info("removing wallet file and app data");
                 application.finalizeWipe();
             }
+            //Clear the blockchain identity
+            executor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    //not sure when this will execute
+                    AppDatabase.getAppDatabase().blockchainIdentityDataDao().clear();
+                    AppDatabase.getAppDatabase().dashPayProfileDao().clear();
+                }
+            });
```

Comment:
When resetting a wallet, this code does not appear to be executed.  Is it because we are using an executor? @tomasz-ludek 

Perhaps the app is restarted before this is executed.  When doing a rescan blockchain, this code is executed.

Reply:
As discussed on Slack. It could be moved to the end of `WalletApplication.finalizeWipe` methode, like this:
```
final Handler mainThreadHandler = new Handler();
Executors.newSingleThreadExecutor().execute(new Runnable() {
    @Override
    public void run() {
        AppDatabase.getAppDatabase().blockchainIdentityDataDao().clear();
        AppDatabase.getAppDatabase().dashPayProfileDao().clear();
        mainThreadHandler.post(new Runnable() {
            @Override
            public void run() {
                ProcessPhoenix.triggerRebirth(WalletApplication.this);
            }
        });
    }
});
```

Reply:
https://github.com/dashevo/dash-wallet/pull/418/commits/abf080a0abd2f695b9c389510137fcd3bdf669e6

this code was working for me